packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,447 @@
+      throw new Error('Cannot specify both `healthCheckConfig` and `healthCheckCustomConfig`.');
+    }
+
+    if (namespaceType !== NamespaceType.DnsPublic && props.healthCheckConfig) {
+      throw new Error('Can only use `healthCheckConfig` for a Public DNS namespace.');
+    }
+
+    if (props.routingPolicy === RoutingPolicy.Multivalue
+        && props.dnsRecordType === DnsRecordType.Cname) {
+      throw new Error('Cannot use `CNAME` record when routing policy is `Multivalue`.');
+    }
+
+    if (props.routingPolicy === RoutingPolicy.Multivalue
+        && props.loadBalancer) {
+      throw new Error('Cannot register loadbalancers when routing policy is `Multivalue`.');
+    }
+
+    if (props.healthCheckConfig
+        && props.healthCheckConfig.type === HealthCheckType.Tcp
+        && props.healthCheckConfig.resourcePath) {
+          throw new Error('Cannot specify `resourcePath` when using a `TCP` health check.');
+    }
+
+    // Set defaults where necessary
+    const routingPolicy = (props.dnsRecordType === DnsRecordType.Cname) || props.loadBalancer
+      ? RoutingPolicy.Weighted
+      : RoutingPolicy.Multivalue;
+
+    const dnsRecordType = props.dnsRecordType || DnsRecordType.A;
+
+    if (props.loadBalancer
+      && (!(dnsRecordType === DnsRecordType.A
+        || dnsRecordType === DnsRecordType.AAAA
+        || dnsRecordType === DnsRecordType.A_AAAA))) {
+      throw new Error('Must support `A` or `AAAA` records to register loadbalancers.');
+    }
+
+    const dnsConfig = props.namespace.type === NamespaceType.Http
+      ? undefined
+      : {
+          dnsRecords: renderDnsRecords(dnsRecordType, props.dnsTtlSec),
+          namespaceId: props.namespace.namespaceId,
+          routingPolicy,
+        };
+
+    const healthCheckConfigDefaults = {
+      type: HealthCheckType.Http,
+      failureThreshold: 1,
+      resourcePath: props.healthCheckConfig && props.healthCheckConfig.type !== HealthCheckType.Tcp
+        ? '/'
+        : undefined
+    };
+
+    const healthCheckConfig = props.healthCheckConfig && { ...healthCheckConfigDefaults, ...props.healthCheckConfig };
+    const healthCheckCustomConfig = props.healthCheckCustomConfig;
+
+    // Create service
+    const service = new CfnService(this, 'Resource', {
+      name: props.name,
+      description: props.description,
+      dnsConfig,
+      healthCheckConfig,
+      healthCheckCustomConfig,
+      namespaceId: props.namespace.namespaceId
+    });
+
+    this.serviceName = service.serviceName;
+    this.serviceArn = service.serviceArn;
+    this.serviceId = service.serviceId;
+    this.namespace = props.namespace;
+    this.dnsRecordType = dnsRecordType;
+    this.routingPolicy = routingPolicy;
+  }
+
+  /**
+   * Registers an ELB as a new instance with unique name instanceId in this service.
+   */
+  public registerLoadBalancer(loadBalancer: route53.IAliasRecordTarget, customAttributes?: {[key: string]: string}): IInstance {
```

Comment:
Referring to my comment above over the `loadBalancer` prop, this method can be called even if `loadBalancer` is set to `false` but will fail later in the validation in `AliasTargetInstance`. Why not centralize load balancer validation in `AliasTargetInstance`?